### PR TITLE
Add code coverage report to unittest workflow

### PR DESCRIPTION
**Pull request by wuweicaifb**
## Prerequisites

- [x] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [x] If this is a non-trivial change, I have already opened an accompanying Issue.
- [x] If applicable, I have included documentation updates alongside my code changes.

## Description

Let's get test coverage automatically on each PR!

## Test Plan

Let's see what will happen in this PR!

### CODE REVIEW

File: .github/workflows/unittest.yml [hunk 1]
```diff
@@ -23,6 +23,10 @@ jobs:
           export CPLUS_INCLUDE_PATH=/usr/include/gdal
           export C_INCLUDE_PATH=/usr/include/gdal
           pip install .
+          pip install coverage
 
       - name: Run All Unittest
-        run: python -m unittest discover terragraph_planner -b
+        run: coverage run -m unittest
```
Comment (dbeylkin) [inline .github/workflows/unittest.yml:29]: Are you replacing the standard unit test run with the coverage run? It seems we might want to have both. Ideally code coverage does not check for correctness (although I imagine the assert checks will force to pass/fail but I wonder if those can be overridden for coverage purposes).

Comment (wuweicaifb) [reply .github/workflows/unittest.yml:29]: Coverage run will run all the tests too. If we have both, we will need double time.

**Comment by wuweicaifb** [comment]
> # [Codecov](https://codecov.io/gh/terragraph/terragraph-planner/pull/2?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=terragraph) Report
> > ❗ No coverage uploaded for pull request base (`main@aecdf65`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=terragraph#section-missing-base-commit).
> > The diff coverage is `n/a`.
> 
> ```diff
> @@           Coverage Diff           @@
> ##             main       #2   +/-   ##
> =======================================
>   Coverage        ?   95.85%           
> =======================================
>   Files           ?      111           
>   Lines           ?    13673           
>   Branches        ?        0           
> =======================================
>   Hits            ?    13106           
>   Misses          ?      567           
>   Partials        ?        0           
> ```
> 
> Help us with your feedback. Take ten seconds to tell us [how you rate us](https://about.codecov.io/nps?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=terragraph). Have a feature suggestion? [Share it here.](https://app.codecov.io/gh/feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=terragraph)

Wow! Good first report!

File: .github/workflows/unittest.yml (hunk 1 above)
Comment (dbeylkin) [reply .github/workflows/unittest.yml:29]: What happens to coverage if a unit test fails? And what happens if the unit test is fixed -- does the coverage report get updated? One question is if coverage should *only* be run once the unit tests all pass.

**Review by dbeylkin — APPROVED** [review]

**Pull request merged**